backend/src/main/java/ai/giskard/domain/Project.java

```diff
@@ -57,7 +57,7 @@ public class Project extends AbstractAuditingEntity {
 
     @Getter
     @Setter
-    @OneToMany(mappedBy = "project", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
+    @OneToMany(mappedBy = "project", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
```

Comment:
Do we have a better way using lazy loading?

Reply:
What error do you get? You probably are outside of a transactional context (`@Transactional`)

Depending on your need you can make a request to fetch the models directly from the repository so that you only fetch them where you need them:

```java
@Query("SELECT p FROM Project p LEFT JOIN FETCH p.models m WHERE p.id = :id")
Project findOneWithModels(long id);
```

Reply:
I got:

```
org.hibernate.LazyInitializationException: failed to lazily initialize a collection of role: ai.giskard.domain.Project.models: could not initialize proxy - no Session
	at org.hibernate.collection.spi.AbstractPersistentCollection.throwLazyInitializationException(AbstractPersistentCollection.java:635)
	at org.hibernate.collection.spi.AbstractPersistentCollection.withTemporarySessionIfNeeded(AbstractPersistentCollection.java:218)
	at org.hibernate.collection.spi.AbstractPersistentCollection.readSize(AbstractPersistentCollection.java:148)
	at org.hibernate.collection.spi.PersistentSet.size(PersistentSet.java:151)
	at ai.giskard.web.rest.controllers.ModelController.createModelMeta(ModelController.java:97)
```

Yes it could be out of a transactional context because there was no DB ops apart from model storage.